client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -456,26 +461,84 @@ def test_batch_predict_with_batches_and_indexes(mocked_responses) -> None:
-                       pd.DataFrame({"D": [31, 32, 33, 34, 35, 36, 37, 38, 39, 40]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       pd.DataFrame({"E": [41, 42, 43, 44, 45, 46, 47, 48, 49, 50]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       ]
+    expected_df_list = [
+        pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"C": [21, 22, 23, 24, 25, 26, 27, 28, 29, 30]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"D": [31, 32, 33, 34, 35, 36, 37, 38, 39, 40]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"E": [41, 42, 43, 44, 45, 46, 47, 48, 49, 50]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        ]
     for expected_df in expected_df_list:
         mocked_responses.add(
             responses.POST,
             BATCH_PREDICTION_URL,
             body=json.dumps(expected_df.to_dict(orient="split")),
             status=200,
-            )
+        )
     creds = EmailCredentials.load_from_os_env()
     dm = DeployedModel(
         prediction_url=PREDICTION_URL,
         creds=creds,
         token=TOKEN,
-        )
+    )
     input_df = pd.DataFrame({"a": [1, 2, 3, 4, 5], "b": [11, 12, 13, 14, 15]}, index=["A", "B", "C", "D", "E"])
     prediction_df = dm.batch_predict(input_df, 1)
     expected_final_df = pd.concat(expected_df_list)
     pd.testing.assert_frame_equal(expected_final_df, prediction_df)
+
+
+@st.composite
+def generate_data(draw):
+    """ Return a dict that represents a dataframe. Generates ints, floats, and strings."""
+    num_rows = draw(st.integers(min_value=50, max_value=200))
+    num_cols = draw(st.integers(min_value=1, max_value=10))
+    col_names = draw(st.lists(st.text(), max_size=num_cols, min_size=num_cols, unique=True))
+    data = {}
+    for name in col_names:
+        type_probability = draw(st.floats(min_value=0, max_value=1))
+        if type_probability <= 0.3:
+            col = draw(st.lists(st.integers(), max_size=num_rows, min_size=num_rows))
+        elif type_probability <= 0.6:
+            col = draw(st.lists(st.floats(), max_size=num_rows, min_size=num_rows))
+        else:
+            col = draw(st.lists(st.text(), max_size=num_rows, min_size=num_rows))
+        data[name] = col
+
+    out_dict = {"data": data}
+    index_probability = draw(st.floats(min_value=0, max_value=1))
+    if index_probability <= 0.5:
+        index = draw(st.lists(st.text(), max_size=num_rows, min_size=num_rows))
+        out_dict["index"] = index
+    return out_dict
+
+
+@given(json_df=generate_data(), batch_size=st.integers(min_value=5, max_value=50))
```

Comment:
How would you feel about going down to 1 here, too?

```suggestion
@given(json_df=generate_data(), batch_size=st.integers(min_value=1, max_value=50))
```